### ISSUE TITLE
Clean up ExonGenomicCoordsMapper

### ISSUE THREAD

**Issue by korikuzma** [GenomicMedLab/cool-seq-tool#224]
A lot of this was written years ago. It's hard to follow what's happening. We should refactor this class

**Comment by korikuzma** [comment]
Apparently it’s hard to follow because the previous model was not the best. There is now a better way to represent using VRS 2.0. @ahwagner your time to shine ✨ I'd also like to throw out that example input/output would be nice 😉

**Comment by korikuzma** [comment]
I think the alignment part makes sense, but have struggled to find a good way to represent what is needed. So going to stop progress until we hear how to represent using VRS 2.0

**Comment by ahwagner** [comment]
Linking the [Cool Seq Tool documentation](https://coolseqtool.readthedocs.io/en/latest/reference/api/mappers/cool_seq_tool.mappers.exon_genomic_coords.html#cool_seq_tool.mappers.exon_genomic_coords.ExonGenomicCoordsMapper) of this method for my own reference.

The idea of this method is to help us go from the ends of fusion transcript segments in exon representation, to where those ends exist on a genomic sequence. On occasion, fusion transcripts contain sequence that is intronic (exists on a genomic sequence but not the transcript), or have a junction that omits some sequence at the end of an exon. In the fusions model, we use [offsets](https://fusions.cancervariants.org/en/latest/information_model.html#structural-elements) to describe this change. This concept of offset representation is also used in [HGVS coding sequence representations](https://hgvs-nomenclature.org/en/latest/background/numbering/#coding-dna-reference-sequences).

I think we should postpone any refactor of this method until the VRS 2.0 beta1 release for the Adjacency class, as there is a fundamental shift from a segment-based model to a junction (adjacency) model. Linking [this thread](https://github.com/ga4gh/vrs/issues/449) for progress on that release. Once that is completed, we should use the beta model to revise how this is represented in FUSOR.

**Comment by korikuzma** [comment]
In slack, @ahwagner asked: I think this is okay to resume now that we have the Adjacency class, right?

@ahwagner You had said in comment about VRS 2.0 beta1, but I don't think we're at beta yet. Did you want us to proceed still?
> I think we should postpone any refactor of this method until the VRS 2.0 beta1 release for the Adjacency class, as there is a fundamental shift from a segment-based model to a junction (adjacency) model.

**Comment by korikuzma** [comment]
Tagging @jsstevenson and @katiestahl so they can follow. Slack doesn't allow thread in a thread

**Comment by ahwagner** [comment]
We did away with Alpha/Beta/RC for maturity levels which may have created some confusion here. I am comfortable with the `draft` model for `Adjacency` from the [latest VRS pre-release](https://github.com/ga4gh/vrs/releases/tag/2.0.0.connect.2024-04.1) and we already have it in VRS-Python, so I think we can move forward on implementing it for Cool-Seq-Tool.

**Comment by korikuzma** [comment]
@ahwagner ah okay. That's my bad for not remembering the [maturity model](https://docs.google.com/document/d/1pqxeB9abRpLe0j6PiRBjmPYPKO9URyBY2QYVcyqO-m8/edit?usp=sharing) changes.

**Comment by ahwagner** [comment]
No fault here. With those changes this issue needed to be clarified. I also don't think anything has been lost by clarifying this now instead of earlier. However, with the recent https://github.com/cancervariants/fusion-curation/issues/277 issue it is a good time to revisit.

**Comment by korikuzma** [comment]
Going to add feedback for changes here: 
- [ ] Rename methods to `genomic_to_tx_segment` and `tx_segment_to_genomic`
- [ ] Strand should not be provided for `genomic_to_tx_segment`
- [ ] In `genomic_to_tx_segment`, rename `start` and `end` to `seg_start` and `seg_end`
- [ ] Assume that coordinates are inter-residue

New structure (Aligned Segment) will look like follows:

```python
{
      "gene": "WEE1",
      "alt_ac": "NC_000011.10",
      "seg_start": {
            "exon_ord": 1,
            "offset": 0,
            "genomic_location": {
                  "type": "SequenceLocation",
                  "sequenceReference": {
                        "type": "SequenceReference",
                        "refgetAccession": "SQ.2NkFm8HK88MqeNkCgj78KidCAXgnsfV1"
                  },
                  "start": 9575887
            }
      },
      "seg_end": {
            "exon_ord": 10,
            "offset": 0,
            "genomic_location": {
                  "type": "SequenceLocation",
                  "sequenceReference": {
                        "type": "SequenceReference",
                        "refgetAccession": "SQ.2NkFm8HK88MqeNkCgj78KidCAXgnsfV1"
                  },
                  "end": 9589767
            }
      },
      "tx_ac": "NM_003390.3"
}
```

@jarbesfeld am I missing anything?

**Comment by korikuzma** [comment]
Fixing some labels 
![Cool-Seq-Tool Mapper(1)](https://github.com/user-attachments/assets/640d26f2-1273-4427-90ec-c08e6092efe1)

**Comment by korikuzma** [comment]
I think all that's left to do in this epic is DRY + smaller methods
